c++/src/capnp/cc_capnp_library.bzl

```diff
@@ -0,0 +1,114 @@
+"""Bazel rule to compile .capnp files into c++."""
+
+capnp_provider = provider("Capnproto Provider", fields = {
+    "includes": "includes for this target (transitive)",
+    "inputs": "src + data for the target",
+    "src_prefix": "src_prefix of the target",
+})
+
+def _workspace_path(label, path):
```

Comment:
why is this needed? shouldn't the path generated by bazel be fine?

Reply:
This is needed to generate -I directives when capnp is an external repo. Then this essentially adds "external/" prefix to paths where needed.

Reply:
Bazel 6 removes the external/ stuff. Given this is a rule I really don't see why the file can't be passed to ctx.actions.args() and expanded automatically by Bazel.

Reply:
> Given this is a rule I really don't see why the file can't be passed to ctx.actions.args() and expanded automatically by Bazel.

This doesn't get files as input, but directories. E.g. src-prefix is not a file but part of the name of the file, etc. I don't see how args will help.

Reply:
Given

```
"_capnp_system": attr.label(default = "@capnp-cpp//src/capnp:capnp_system_library"),
```

Can't this be changed to `attr.label(allow_files = True)` and then use `ctx.files._capnp_system[0].dirname`? I just don't see how appending the workspace root makes any sense within an action.

Reply:
src_prefix is not like this anyway.

Reply:
but done for capnp_system